receiver/otlpreceiver/otlphttp.go

```diff
@@ -42,7 +43,7 @@ func handleTraces(resp http.ResponseWriter, req *http.Request, tracesReceiver *t
 
 	otlpResp, err := tracesReceiver.Export(req.Context(), otlpReq)
 	if err != nil {
-		writeError(resp, enc, err, http.StatusInternalServerError)
+		writeError(resp, enc, err, errors.GetHTTPStatusCodeFromStatus(err))
```

Comment:
I don't like that now we also have a similar but different logic in `errorMsgToStatus`. Can we consolidate?

Reply:
Agreed. I took a look at consolidating before opening the PR. I believe it can be refactored, but it caused a lot of changes unrelated to the issue I was trying to solve. To keep the PRs smaller and targeted I'd like to use this PR as a solution to the issue and then a future PR that is only a refactor. I can open an issue to track that refactor if you'd like.

Reply:
@bogdandrutu are you ok with that approach?

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector/issues/9864